lib/grape/validations/validators/coerce.rb

```diff
@@ -56,7 +56,14 @@ def coerce_value(type, val)
         return val || Set.new if type == Set
         return val || {}      if type == Hash
 
-        converter = Virtus::Attribute.build(type)
+        # To support custom types that Virtus can't easily coerce, pass in an
+        # explicit coercer. Custom types must implement a `parse` class method.
+        converter_options = {}
+        if ParameterTypes.custom_type?(type)
+          converter_options[:coercer] = type.method(:parse)
```

Comment:
Nitpick: if ParameterTypes knows `custom_type?` it should also have `ParameterTypes.custom_type_parser_for(type)` or something like that, otherwise there's an assumption that custom_type means parse exists in two places.